app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/SessionsRemoteDataSource.java

```diff
@@ -35,17 +37,16 @@ public SessionsRemoteDataSource(DroidKaigiClient client) {
                         }
                     }
                     return sessions;
-                })
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread());
+                });
     }
 
     @Override
     public Maybe<Session> find(int sessionId, String languageId) {
-        return findAll(languageId).map(sessions -> Stream.of(sessions)
+        return findAll(languageId)
+                .toObservable()
+                .flatMap(Observable::fromIterable)
                 .filter(session -> session.id == sessionId)
-                .findSingle()
-                .get()).toMaybe();
+                .singleElement();
```

Comment:
It crashes if there is no value in the original implementation. So I replaced it with flatMap.